provisioning_client/adapters/hsm_client_riot.c

```diff
@@ -551,6 +548,15 @@ void hsm_client_riot_destroy(HSM_CLIENT_HANDLE handle)
         /* Codes_SRS_HSM_CLIENT_RIOT_07_008: [ hsm_client_riot_destroy shall free the HSM_CLIENT_HANDLE instance. ] */
         free(x509_client->certificate_common_name);
         /* Codes_SRS_HSM_CLIENT_RIOT_07_009: [ hsm_client_riot_destroy shall free all resources allocated in this module. ] */
+        mbedtls_ecp_point_free(&x509_client->ca_root_pub);
```

Comment:
Please extract a function here that will free an x509 certificate (by taking the pub and pri keys) and call 3 times.
That way it's far harder to forget to release a portion. (Ideally the structs should have had the same interface.)

Reply:
That is what I had originally and will work, I took Dennis's suggestion on calling that group free API, this will require a rewrite of the UT

Reply:
Fixed